examples/ota-provider-app/linux/UserConsentProvider.h

```diff
@@ -0,0 +1,62 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+#include <ota-provider-common/UserConsentDelegate.h>
+
+class UserConsentProvider : public chip::ota::UserConsentDelegate
+{
+public:
+    void ObtainUserConsentAsync(chip::NodeId nodeId, chip::EndpointId endpoint, uint32_t currentVersion, uint32_t newVersion)
+    {
+        mNodeId         = nodeId;
+        mEndpoint       = endpoint;
+        mCurrentVersion = currentVersion;
+        mNewVersion     = newVersion;
+
+        // This is not an async operation, std::cin blocks the chip thread
+        // TODO: Spawn a new thread for getting user consent
+        chip::DeviceLayer::SystemLayer().StartTimer(chip::System::Clock::Milliseconds32(1 * 1000), &HandleUserConsent, this);
```

Comment:
Note optionally, you can write `Clock::Seconds32(1)` and the constant is converted at compile time with no code cost.